### ISSUE TITLE
Improvements to anntzer/vim-cython?

### ISSUE THREAD

**Issue by anntzer** [lambdalisue/vim-cython-syntax#1]
Hi,
I saw that you forked my repo (https://github.com/anntzer/vim-cython) a few days ago but then decided to switch to a brand new implementation.  Is there any reason for this choice?  Are there any specific issues on my implementation that you'd like to see fixed?
A.

**Comment by lambdalisue** [comment]
Yes. I used your version but noticed that the syntax highlight is a bit different from Vim's python syntax. I guessed that you once copied the syntax and modified right? The Python syntax has modified recently and I guessed that why the syntax is a bit different.
I first tried to fork your version but it was a bit hard to fix your version so I decide to make a new one with overriding the existing Python syntax instead of copy so that any modification on Python syntax will be reflected.

https://github.com/lambdalisue/vim-cython-syntax/blob/master/syntax/cython.vim

Additionally, the following points were not fit me
- `ftdetect` force to assign `cython` so that I cannot use `ft=diff` or so on
- `ftplugin` assign some unwilling mappings. I just needed a simple one.

**Comment by anntzer** [comment]
The syntax file is based on https://github.com/hdima/python-syntax, which is itself based on the original syntax file shipped by Vim (some time ago).  Most of the additions (compared to Vim's original syntax file) should be switchable by flags.

I don't really understand your use-case for ft=diff; in any case you'll need to set it back after opening the file, no?  (The one-liner I use is pretty standard I think?)  The mappings in ftplugin could be made switchable too.

In any case, I'd rather work with you to make a single plugin as good as possible, and am very open to PRs.

**Comment by lambdalisue** [comment]
>  Most of the additions (compared to Vim's original syntax file) should be switchable by flags.

No, the point what I didn't like is that the syntax file copied everything so the modification on the original Python syntax file could not be reflected. That's why my implementation tried to use the existing one as much as possible. I would love to rely on the default Python syntax file.

> I don't really understand your use-case for ft=diff; in any case you'll need to set it back after opening the file, no?

It is a bit hard to explain but `ftdetect/xxx.vim` is seemed be called after `BufReadCmd` autocmd so `setlocal filetype=xxx` would overwrite `filetype=diff` assigned by `BufReadCmd`. Mean that plugin which use `BufReadCmd` to open a pseudo buffer (e.g. `http://example.com/example.pyx`) with `filetype=diff` would open `filetype=cython` instead of `filetype=diff`.

>  The mappings in ftplugin could be made switchable too.

Yes. That's why the mapping in ftplugin is not a reason why I made a new one.

> In any case, I'd rather work with you to make a single plugin as good as possible, and am very open to PRs.

Thanks. But I'm not really sure if it is possible while I prefer to rely on the default Python syntax file but it seems you would like to assign everything from scratch. I feel it is an opposite favour.

**Comment by anntzer** [comment]
Indeed, I see that you first call `runtime! syntax/python.vim` then layer your changes on top of that.  While this is in theory very nice, this will run into all kinds of issues if someone uses a different implementation of `python.vim`.  For a similar issue, see hdima/python-syntax#30.

Basically, I think that with the current stage of vim plugins packaging (no dependency management), depending on a file that you don't control is too brittle (on the other hand, I just realized that there may be a solution involving git shallow submodules... perhaps I'll give it a try).

In any case, good luck :-)

**Comment by lambdalisue** [comment]
> Indeed, I see that you first call runtime! syntax/python.vim then layer your changes on top of that. While this is in theory very nice, this will run into all kinds of issues if someone uses a different implementation of python.vim. For a similar issue, see hdima/python-syntax#30.

Good point. Additionally, your implementation has an advantage of the performance as well while mine has redundancy of definition. But even so, I felt that I would like to rely on the existing one while it could make the code simple and easy to modify.

> In any case, good luck :-)

You too thanks ;-)
